### PR TITLE
Add Parallel Gateway

### LINKED ISSUE #195: Parallel Gateway
As a designer I would like a Parallel Gateway to represent two concurrent tasks in workflow.

https://www.lucidchart.com/pages/bpmn-gateway-types?a=1#section_2

https://docs.camunda.org/manual/7.8/reference/bpmn20/gateways/parallel-gateway/

### PR DESCRIPTION
**Pull request by tuantruong625**
Fixes #195 

Shape with appropriate attributes are included
- id
- name
- gatewayDirection 

Validation will be added as a separate feature.

- require a merge gateway when a parallel gateway is added

_As per @velkymx's comment in https://github.com/ProcessMaker/modeler/issues/195#issuecomment-461488974_

### CODE REVIEW

File: src/components/nodes/parallelGateway/index.js [hunk 1]
```diff
@@ -48,6 +50,18 @@ export default {
...
         },
+        {
+          component: 'FormSelect',
+          config: {
+            label: 'Direction',
+            helper: 'Select direction of gateway',
+            name: 'gatewayDirection',
+            options: [
+              { value: `${ gatewayDirection.diverging }`, content: 'Diverging' },
+              { value: `${ gatewayDirection.converging }`, content: 'Converging' },
```
Comment (chipit24) [inline src/components/nodes/parallelGateway/index.js]: Same here re: template strings.

File: src/components/nodes/gateway/gatewayConfig.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const gatewayDirection = { diverging: 'diverging', converging: 'converging' };
```
Comment (chipit24) [inline src/components/nodes/gateway/gatewayConfig.js]: It seems you export almost the same object in `src/components/nodes/gateway/index.js`:
```js
export const gatewayDirectionOptions = { Diverging: 'Diverging', Converging: 'Converging' };
```

I think this config can be removed from one of these spots.

File: src/components/nodes/parallelGateway/parallelGateway.vue [hunk 3]
```diff
@@ -4,93 +4,13 @@
...
-      ],
-    };
-  },
-  watch: {
-    'node.definition.name'(name) {
-      const { width } = this.shapeView.getBBox();
-      this.shape.attr('.label/text', joint.util.breakText(name, { width }));
-    },
-  },
+  extends: GatewayComponent,
```
Comment (chipit24) [inline src/components/nodes/parallelGateway/parallelGateway.vue]: I think you can do the same refactor (extend) in `inclusiveGateway.vue` as well.

**Comment by tuantruong625** [comment]
Do you mean how the sequence flow overlaps the task element ? If so this applies to the other gateways as well, and we should create a separate issue for it.

**Comment by velkymx** [comment]
> Do you mean how the sequence flow overlaps the task element ? If so this applies to the other gateways as well, and we should create a separate issue for it.

The lines should not overlap. Do you think this looks correct?

**Comment by velkymx** [comment]
Should look more like this:

screenshot 2019-02-12 14 28 52

**Comment by chipit24** [comment]
@velkymx The issue with sequence flows overlapping other elements occurs not just with parallel gateways, but with all flow elements. Previously, we used the [manhattan router](https://resources.jointjs.com/docs/jointjs/v2.2/joint.html#routers.manhattan) in JointJS, which offered obstacle avoidance, but this created other issues.

We created an issue with sequence flows overlapping elements here: https://github.com/ProcessMaker/modeler/issues/183. I think it makes sense to tackle that issue in that ticket and create another PR for that, since it's not directly related to this PR.

Looking at your updated image, we could break the issue down into first modifying flows coming out of gateways. Would be interesting to set up, since there are still edge cases we need to address. Might be easier to let the user click and drag the flows.

**Pull request merged**